<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>10X 单细胞测序数据分析 • BioCAsia2020scRNAseqWorkshop</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="10X 单细胞测序数据分析">
<meta property="og:description" content="BioCAsia2020scRNAseqWorkshop">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioCAsia2020scRNAseqWorkshop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/sc_analysis.html">10X 单细胞测序数据分析</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/YOU-k/scRNA-seq-workshop">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="sc_analysis_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>10X 单细胞测序数据分析</h1>
                        <h4 class="author">Luyi Tian</h4>
                                    <h4 class="author">Yue You</h4>
                        
      
      <small class="dont-index">Source: <a href="https://github.com/YOU-k/scRNA-seq-workshop/blob/master/vignettes/sc_analysis.Rmd"><code>vignettes/sc_analysis.Rmd</code></a></small>
      <div class="hidden name"><code>sc_analysis.Rmd</code></div>

    </div>

    
    
<div id="description" class="section level2">
<h2 class="hasAnchor">
<a href="#description" class="anchor"></a>Description</h2>
<p>In this workshop (presented in Mandarin), you will learn how to analyse single-cell RNA-sequencing count data produced by the Chromium 10x platform using R/Bioconductor. This will include reading the data into R, pre-processing data, normalization, feature selection, dimensionality reduction and downstream analysis, such as clustering and cell type annotation.</p>
<p>Expectation: You will learn how to generate common plots for analysis and visualisation of single cell gene expfression data, such as diagnostic plots to assess the data quality as well as dimensionality reduction techniques such as principal components analysis and t-distributed stochastic neighbourhood embedding (t-SNE). The material we will be covering on single-cell RNA-sequencing analysis is a subset of the work of Amerzquita et al. (2020) Nature Methods,17:137–145 available at <a href="https://osca.bioconductor.org" class="uri">https://osca.bioconductor.org</a>.</p>
<p>Pre-requisites: The course is aimed at PhD students, Master’s students, and third &amp; fourth year undergraduate students. Some basic R knowledge is assumed - this is not an introduction to R course. If you are not familiar with the R statistical programming language it is compulsory that you work through an introductory R course before you attend this workshop.</p>
<div id="participation" class="section level3">
<h3 class="hasAnchor">
<a href="#participation" class="anchor"></a>Participation</h3>
<p>After the lecture, participants are expected to follow along the hands-on session. we highly recommend participants bringing your own laptop.</p>
</div>
<div id="r-bioconductor-packages-used" class="section level3">
<h3 class="hasAnchor">
<a href="#r-bioconductor-packages-used" class="anchor"></a><em>R</em> / <em>Bioconductor</em> packages used</h3>
<p>The following R/Bioconductor packages will be explicitly used:</p>
<ul>
<li>DropletUtils</li>
<li>scran</li>
<li>scater</li>
</ul>
</div>
<div id="time-outline" class="section level3">
<h3 class="hasAnchor">
<a href="#time-outline" class="anchor"></a>Time outline</h3>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Introduction to scRNA-seq</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Analysis workflow</td>
<td>20m</td>
</tr>
<tr class="odd">
<td>Hands on session</td>
<td>15m</td>
</tr>
<tr class="even">
<td>Q &amp; A</td>
<td>10m</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="数据准备" class="section level2">
<h2 class="hasAnchor">
<a href="#%E6%95%B0%E6%8D%AE%E5%87%86%E5%A4%87" class="anchor"></a>数据准备</h2>
<p>下载10x PBMC4k数据并读取。</p>
<div id="质量控制" class="section level3">
<h3 class="hasAnchor">
<a href="#%E8%B4%A8%E9%87%8F%E6%8E%A7%E5%88%B6" class="anchor"></a>质量控制。</h3>
<p>由于10x 技术会使得外部RNA，也就是不包含微滴的里的细胞被同时测序。在分析之前，我们需要确保得到的barcode数据都相对应活细胞。 这里，我们使用<code>emptyDrops</code>来筛选“真”细胞。</p>
<p>对于得到的基因，根据gene ID 进行基因注释，得到基因名，并且知道其所在染色体（用于之后的细胞质量控制）。</p>
<p>这里，我们认为线粒体表达量高的barcode对应细胞质mRNA已经流出的破损细胞。并用这个指标来进行筛查过滤。</p>
<pre><code>##    Mode   FALSE    TRUE 
## logical    3985     315</code></pre>
<p><img src="sc_analysis_files/figure-html/unnamed-chunk-7-1.png" width="768"></p>
<p><img src="sc_analysis_files/figure-html/unnamed-chunk-8-1.png" width="768"></p>
</div>
<div id="数据标准化" class="section level3">
<h3 class="hasAnchor">
<a href="#%E6%95%B0%E6%8D%AE%E6%A0%87%E5%87%86%E5%8C%96" class="anchor"></a>数据标准化</h3>
<p>由于在mRNA分子的捕获，逆转录，测序过程中存在一定的技术误差，相同细胞的计数深度可能不相同。我们需要在进行细胞之间的比对之前，先尽可能的排除采样效应。进行数据标准化，使得细胞之间存在可比性。这里我们使用<code>scran</code>(Lun et al,2016a)，一种允许更多细胞异质性的方法。</p>
<pre><code>##     Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
##  0.00749  0.71207  0.87490  1.00000  1.09900 12.25412</code></pre>
<p><img src="sc_analysis_files/figure-html/unnamed-chunk-11-1.png" width="768"></p>
</div>
<div id="特征选择" class="section level3">
<h3 class="hasAnchor">
<a href="#%E7%89%B9%E5%BE%81%E9%80%89%E6%8B%A9" class="anchor"></a>特征选择</h3>
<p>为了减轻下游分析工具的计算负担，减少数据中的噪声，我们需要先进行特征选择。</p>
<p><img src="sc_analysis_files/figure-html/unnamed-chunk-13-1.png" width="768"></p>
<p>特征选择后，单细胞表达矩阵的维数可以通过专门的降维算法进一步降低。使得数据可以直观可视化，并且将数据简化为基本组成部分。</p>
<pre><code>## [1] 9</code></pre>
</div>
</div>
<div id="聚类分析" class="section level2">
<h2 class="hasAnchor">
<a href="#%E8%81%9A%E7%B1%BB%E5%88%86%E6%9E%90" class="anchor"></a>聚类分析</h2>
<div id="使用louvain算法进行聚类" class="section level3">
<h3 class="hasAnchor">
<a href="#%E4%BD%BF%E7%94%A8louvain%E7%AE%97%E6%B3%95%E8%BF%9B%E8%A1%8C%E8%81%9A%E7%B1%BB" class="anchor"></a>使用<code>louvain</code>算法进行聚类</h3>
<p><code>louvain</code>算法是一个基于图的聚类算法，我们首先构建Shared Nearest Neighbor（SNN）图，输入为使用主成分分析（PCA）降维后得到的矩阵。 在得到SNN图<code>g</code>之后，我们使用它作为<code>louvain</code>算法的输入。</p>
<pre><code>## clust
##   1   2   3   4   5   6   7   8   9  10  11  12  13  14 
## 353 152 376 857  48  46  68 661 287  53 541 198 219 126</code></pre>
<p>使用tSNE对聚类进行可视化。</p>
<p><img src="sc_analysis_files/figure-html/unnamed-chunk-17-1.png" width="768"></p>
</div>
<div id="寻找聚类特异表达基因" class="section level3">
<h3 class="hasAnchor">
<a href="#%E5%AF%BB%E6%89%BE%E8%81%9A%E7%B1%BB%E7%89%B9%E5%BC%82%E8%A1%A8%E8%BE%BE%E5%9F%BA%E5%9B%A0" class="anchor"></a>寻找聚类特异表达基因</h3>
<p>在获得了聚类之后，我们想要知道每个聚类都哪些高表达的基因，这些基因往往是能区分不同细胞类型的marker gene，也可以帮助我们了解不同聚类的生物学功能。</p>
<p>这里，我们通过比对某一cluster和其他所有cluster的表达来找到marker gene， 并且只挑选显示上调基因。</p>
<pre><code>## DataFrame with 33694 rows and 16 columns
##                   p.value       FDR summary.logFC    logFC.2     logFC.3
##                 &lt;numeric&gt; &lt;numeric&gt;     &lt;numeric&gt;  &lt;numeric&gt;   &lt;numeric&gt;
## LTB           4.74177e-07 0.0159769      0.364873   1.727152    2.003863
## TRADD         3.34176e-05 0.5629862      0.246261   0.246261    0.358697
## TNFAIP3       3.64851e-04 1.0000000      0.294551   0.586788    0.324352
## RP11-138A9.1  5.51754e-04 1.0000000      0.235830   0.265198    0.204199
## CD69          9.05928e-04 1.0000000      0.220119   0.889346    0.220119
## ...                   ...       ...           ...        ...         ...
## RP11-1143G9.4           1         1      -2.88441  0.0632559  0.04136662
## CD79B                   1         1      -2.65518 -0.0319757  0.04062722
## CST3                    1         1      -3.38955  0.1132112 -0.00139546
## TYROBP                  1         1      -3.77926 -0.2135870 -0.12432406
## CD79A                   1         1      -2.88958  0.0547465  0.02026294
##                   logFC.4    logFC.5    logFC.6    logFC.7    logFC.8
##                 &lt;numeric&gt;  &lt;numeric&gt;  &lt;numeric&gt;  &lt;numeric&gt;  &lt;numeric&gt;
## LTB              0.709584   2.236589   3.179486   1.384048   3.011681
## TRADD            0.455837   0.399715   0.625041   0.340638   0.494083
## TNFAIP3          0.386801   0.634728   0.640136   0.539046   0.355789
## RP11-138A9.1     0.297797   0.338809   0.459562   0.235830   0.436762
## CD69             0.182859   1.453419   1.796272   0.809607   1.625559
## ...                   ...        ...        ...        ...        ...
## RP11-1143G9.4  0.02079562 -0.0410564 -0.0987084 -0.0922961 -2.8844113
## CD79B          0.03988754 -0.0332708  0.0275658 -0.2155062  0.0970271
## CST3          -0.00267045 -2.1438978 -1.5534313 -0.1021948 -3.3895523
## TYROBP        -0.04510738 -1.6678515 -0.4775968  0.0929494 -3.7792602
## CD79A         -0.02744100 -0.1171942  0.0907275 -0.0144729  0.0329842
##                  logFC.9  logFC.10   logFC.11   logFC.12    logFC.13   logFC.14
##                &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;  &lt;numeric&gt;   &lt;numeric&gt;  &lt;numeric&gt;
## LTB             0.364873  1.564212   0.345040   2.919553    2.640417   2.967425
## TRADD           0.219922  0.466701   0.454696   0.485817    0.455422   0.358617
## TNFAIP3         0.573474  0.294551   0.641317   0.537594    0.548100   0.548964
## RP11-138A9.1    0.261783  0.382809   0.292954   0.393811    0.414021   0.436237
## CD69            1.210379  1.001879   0.441418   0.696360    1.577163   1.468100
## ...                  ...       ...        ...        ...         ...        ...
## RP11-1143G9.4  0.0667527 -2.190784  0.0273067  0.0288028 -1.07943585 -1.7809210
## CD79B         -0.0434586 -0.464497 -2.6551786 -0.0206008 -0.27267278 -0.0135700
## CST3          -0.0285401 -2.688015  0.0152875  0.1102020 -3.64978261 -4.3511281
## TYROBP        -0.0414366 -3.077268 -0.0436713 -2.3635000 -3.65373554 -2.3592519
## CD79A          0.0118844 -0.347398 -2.8895764  0.0677690 -0.00864517 -0.0389843</code></pre>
<p>查看marker gene。</p>
<pre><code>##  [1] "LTB"          "TRADD"        "TNFAIP3"      "RP11-138A9.1" "CD69"        
##  [6] "RORA"         "KLRG1"        "SESN1"        "TRGC1"        "LINC00987"   
## [11] "DUSP2"        "CD8A"         "KLRC2"        "CRTAM"        "GADD45B"     
## [16] "CCR7"         "RPS13"        "RPL35A"       "RPL34"        "RPL21"       
## [21] "ITM2C"        "LILRA4"       "IRF7"         "JCHAIN"       "MZB1"        
## [26] "PF4"          "SDPR"         "TMSB4X"       "NRGN"         "TAGLN2"      
## [31] "TAP1"         "DUSP4"        "CTLA4"        "FOXP3"        "PTTG1"       
## [36] "S100A6"       "LYZ"          "CSTA"         "FCN1"         "S100A9"      
## [41] "GIMAP7"       "MALAT1"       "RPLP1"        "GIMAP1"       "GCC2"        
## [46] "MBNL1"        "NDUFB11"      "NUFIP2"       "ALPK1"        "ZSWIM8"      
## [51] "CD79A"        "HLA-DOB"      "VPREB3"       "CD72"         "CD79B"       
## [56] "KLRF1"        "PRF1"         "SPON2"        "FGFBP2"       "GNLY"        
## [61] "LST1"         "IFITM3"       "AIF1"         "MS4A7"        "SERPINA1"    
## [66] "HLA-DRB1"     "HLA-DQA1"     "HLA-DPB1"     "HLA-DRA"      "HLA-DPA1"</code></pre>
<p>查看特定基因在不同cluster的表达量。</p>
<p><img src="sc_analysis_files/figure-html/unnamed-chunk-22-1.png" width="768"></p>
<p>使用热图对marker gene进行可视化。</p>
<p><img src="sc_analysis_files/figure-html/unnamed-chunk-23-1.png" width="768"></p>
</div>
</div>
<div id="使用singler进行细胞类型注释" class="section level2">
<h2 class="hasAnchor">
<a href="#%E4%BD%BF%E7%94%A8singler%E8%BF%9B%E8%A1%8C%E7%BB%86%E8%83%9E%E7%B1%BB%E5%9E%8B%E6%B3%A8%E9%87%8A" class="anchor"></a>使用<code>singleR</code>进行细胞类型注释</h2>
<p>我们选择 BlueprintEncodeData() 来作为参考数据。</p>
<pre><code>## 
##      B-cells CD4+ T-cells CD8+ T-cells           DC  Eosinophils Erythrocytes 
##          549          773         1274            1            1            5 
##          HSC    Monocytes     NK cells 
##           14         1117          251</code></pre>
<p><img src="sc_analysis_files/figure-html/unnamed-chunk-26-1.png" width="768"></p>
<p>利用每个cluster和对每个细胞计算得到的可能细胞类型画heatmap。</p>
<p><img src="sc_analysis_files/figure-html/unnamed-chunk-27-1.png" width="768"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Yue You, Luyi Tian.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
